src/transport/builder/umessagebuilder.rs

```diff
@@ -0,0 +1,733 @@
+    /// use up_rust::transport::builder::UMessageBuilder;
+    ///
+    /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+    /// let destination = UUri::try_from("my-vehicle/cabin/1/doors")?;
+    /// let message = UMessageBuilder::notification(&destination).build()?;
+    /// assert_eq!(message.attributes.type_, UMessageType::UMESSAGE_TYPE_PUBLISH.into());
+    /// assert_eq!(message.attributes.priority, UPriority::UPRIORITY_CS1.into());
+    /// assert_eq!(message.attributes.sink, Some(destination).into());
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn notification(destination: &'a UUri) -> UMessageBuilder<'a> {
+        UMessageBuilder {
+            validator: Box::new(crate::transport::validator::PublishValidator),
+            message_type: UMessageType::UMESSAGE_TYPE_PUBLISH,
+            sink: Some(destination),
+            ..Default::default()
+        }
+    }
+
+    /// Gets a builder for creating an RPC *request* message.
+    ///
+    /// A request message is used to invoke a service's method with some input data, expecting
+    /// the service to reply with a correlated response message, which is set using the
+    /// [`UMessageBuilder::with_request_id`] function.
+    ///
+    /// # Arguments
+    ///
+    /// * `method_to_invoke` - The URI identifying the method to invoke.
+    /// * `reply_to_address` - The URI that the sender of the request expects the response message at.
+    /// * `ttl` - The number of milliseconds after which the request should no longer be processed
+    ///           by the target service. The value is capped at [`i32::MAX`].
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use up_rust::uprotocol::SerializationError;
+    /// use up_rust::uprotocol::{UMessageType, UPriority, UUri};
+    /// use up_rust::transport::builder::UMessageBuilder;
+    ///
+    /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+    /// let method_to_invoke = UUri::try_from("my-vehicle/cabin/1/rpc.unlock_doors")?;
+    /// let reply_to_address = UUri::try_from("my-cloud/dashboard/1/rpc.response")?;
+    /// let message = UMessageBuilder::request(&method_to_invoke, &reply_to_address, 5000).build()?;
+    /// assert_eq!(message.attributes.type_, UMessageType::UMESSAGE_TYPE_REQUEST.into());
+    /// assert_eq!(message.attributes.priority, UPriority::UPRIORITY_CS1.into());
+    /// assert_eq!(message.attributes.source, Some(reply_to_address).into());
+    /// assert_eq!(message.attributes.sink, Some(method_to_invoke).into());
+    /// assert_eq!(message.attributes.ttl, Some(5000));
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn request(
+        method_to_invoke: &'a UUri,
+        reply_to_address: &'a UUri,
+        ttl: u32,
+    ) -> UMessageBuilder<'a> {
+        UMessageBuilder {
+            validator: Box::new(crate::transport::validator::RequestValidator),
+            message_type: UMessageType::UMESSAGE_TYPE_REQUEST,
+            source: Some(reply_to_address),
```

Comment:
Also related to the [issue I made](https://github.com/eclipse-uprotocol/up-spec/issues/68).